app/services/foreman/password_hash.rb

```diff
@@ -9,12 +9,31 @@
 #
 module Foreman
   class PasswordHash
+    class PBKDF2Implementation
+      SIZE = 39
```

Comment:
Can you comment on what this size is? I see `generate_salt` accepts a size parameter but `hash_secret` doesn't.

Reply:
Renamed to much more reasonable `DEFAULT_SALT_LENGTH`.